src/can.c

```diff
@@ -449,7 +449,7 @@ static void thingset_can_reqresp_timeout_handler(struct k_timer *timer)
 {
     struct thingset_can_request_response *rr =
         CONTAINER_OF(timer, struct thingset_can_request_response, timer);
-    rr->callback(NULL, 0, 0, -ETIMEDOUT, 0, rr->cb_arg);
+    rr->callback(NULL, 0, 0, -ETIMEDOUT, rr->can_id, rr->cb_arg);
```

Comment:
The `source_addr` parameter of the callback is of type `uint8_t`, but the CAN ID is of type `uint32_t`. We should extract the address from the ID.